exercises/calculator.cpp

```diff
@@ -10,3 +10,74 @@
   Multiplication: 8
   Division: 2
 */
+
+#include <iostream>
+#include <string>
+
+const std::string ERROR_COLOR = "\033[31m";
+const std::string NORMAL_COLOR = "\033[0m";
+
+const int ZERO = 48;
+const int NINE = 57;
+const int MAX_DIGIT = 10;
+const int MINUS = 45;
+
+int insertNumber(std::string position);
+bool checkStringIsNumber(std::string in);
+
+float doSum(float n1, float n2);
+float doSub(float n1, float n2);
+float doMolt(float n1, float n2);
+float doDiv(float n1, float n2);
+
+void doOperation(float n1, float n2);
+
+int main(){
+  float num1 = insertNumber("first");
+  float num2 = insertNumber("second");
+
+  doOperation(num1, num2);
+}
+
+
+
+int insertNumber(std::string position){
+  std::string insert;
+  bool isCorrect = true;
+  do {
+    if(!isCorrect) std::cout << ERROR_COLOR+("[Insert a valid input] ")+NORMAL_COLOR;
+    std::cout << "Insert "<< position <<" number: ";
+    getline(std::cin, insert);
+  } while(!(isCorrect = checkStringIsNumber(insert)));
+
+  return std::stoi(insert);
+}
+
+bool checkStringIsNumber(std::string in){
+  if(in.empty()) return false;
+  if(in.size() > MAX_DIGIT) return false;
+  for(int i = 0; i < in.size(); i++)
+    if((in.at(i) < ZERO || in.at(i) > NINE) && in.at(i) != MINUS) return false;
+
+  return true;
+}
```

Comment:
Gli spazi non costano, puoi usarli ;-)
```suggestion
bool checkStringIsNumber(std::string in){
  if (in.empty())
      return false;

  if (in.size() > MAX_DIGIT)
      return false;

  for (int i = 0; i < in.size(); i++)
    if ((in.at(i) < ZERO || in.at(i) > NINE) && in.at(i) != MINUS)
        return false;

  return true;
}
```